lib/ruby_lsp/requests/semantic_highlighting.rb

```diff
@@ -339,38 +391,47 @@ def add_token(location, type, modifiers = [])
 
       private
 
-      # Exclude the ":" symbol at the end of a location
-      # We use it on keyword parameters to be consistent
-      # with the rest of the parameters
-      sig { params(location: T.untyped).returns(SyntaxTree::Location) }
-      def location_without_colon(location)
-        SyntaxTree::Location.new(
-          start_line: location.start_line,
-          start_column: location.start_column,
-          start_char: location.start_char,
-          end_char: location.end_char - 1,
-          end_column: location.end_column - 1,
-          end_line: location.end_line,
-        )
-      end
-
-      # Textmate provides highlighting for a subset
-      # of these special Ruby-specific methods.
-      # We want to utilize that highlighting, so we
-      # avoid making a semantic token for it.
+      # Textmate provides highlighting for a subset of these special Ruby-specific methods.  We want to utilize that
+      # highlighting, so we avoid making a semantic token for it.
       sig { params(method_name: String).returns(T::Boolean) }
       def special_method?(method_name)
         SPECIAL_RUBY_METHODS.include?(method_name)
       end
 
-      sig { params(value: SyntaxTree::Ident).returns(Symbol) }
-      def type_for_local(value)
-        local = @emitter.current_scope.find_local(value.value)
+      sig { params(node: YARP::CallNode).void }
+      def process_regexp_locals(node)
+        receiver = node.receiver
+
+        # The regexp needs to be the receiver of =~ for local variable capture
+        return unless receiver.is_a?(YARP::RegularExpressionNode)
+
+        content = receiver.content
+        loc = receiver.content_loc
+
+        # For each capture name we find in the regexp, look for a local in the current_scope
+        Regexp.new(content, Regexp::FIXEDENCODING).names.each do |name|
+          # The +3 is to compensate for the "(?<" part of the capture name
+          capture_name_offset = T.must(content.index("(?<#{name}>")) + 3
```

Comment:
Not a blockers, but it seems there are some other ways of representing named captures:

https://rubyapi.org/3.2/o/regexp#method-i-named_captures:~:text=Named%20groups%20can%20be%20backreferenced%20with%20%5Ck%3Cname%3E%2C%20where%20name%20is%20the%20group%20name.

Reply:
I think the `\k<>` is a backreference to the named capture and not another way to create named captures. The idea is for you to do this

```ruby
/(?<group>(something|other))(\k<group>)/
#  ^ this captures group       ^ this references the captured group
```